cmd/kube-apiserver/app/server.go

```diff
@@ -484,6 +486,17 @@ func buildGenericConfig(
 	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIPriorityAndFairness) && s.GenericServerRunOptions.EnablePriorityAndFairness {
 		genericConfig.FlowControl, lastErr = BuildPriorityAndFairness(s, clientgoExternalClient, versionedInformers)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIServerIdentity) && utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StorageVersionAPI) &&
+		utilfeature.DefaultFeatureGate.Enabled(genericfeatures.UnknownVersionInteroperabilityProxy) {
+		if s.SecureServing.ServerCert.CertKey.RootCAFile != "" {
+			genericConfig.UnknownVersionProxy, lastErr = BuildUnknownVersionProxy(versionedInformers, genericConfig.StorageVersionManager, s.ProxyClientCertFile,
+				s.ProxyClientKeyFile, s.SecureServing.ServerCert.CertKey.RootCAFile)
+
+		} else {
+			lastErr = fmt.Errorf("No file specified for root-ca-file")
```

Comment:
Acknowledged, this definitely should not be a breaking change. I think there would be a self signed CA cert generated  if this flag is not provided (I have added code to do this), so perhaps we don't need this check. I will double check and update.

Reply:
Generation probably won't work here since this is only useful in HA settings where the other apiservers need to use this as the root of the chain of trust for their respective client certs.

Reply:
I don't think we can use the `--client-ca-file` file to verify server certs. This file is meant to be used to verify client certs IIUC.. also just now tried to use the client-ca-file for the proxy, it fails with `tls: failed to verify certificate: x509: certificate signed by unknown authority`

Reply:
if you don't give the API server the new information, you don't get the peer-to-peer proxying behavior

Reply:
> I don't think that's possible, the API server doesn't have the information it needs today

We should update the KEP goals to match.

Reply:
Noted.